// dico-pokemon/dico-pokemon.js
var getUltimatePokemon = function(n)
{
    return ultimate[n];
}
exports.getUltimatePokemon = getUltimatePokemon;

var getVeryRarePokemon = function(n)
{
    return very_rare[n];
}
exports.getVeryRarePokemon = getVeryRarePokemon;

var getRarePokemon = function(n)
{
    return rare[n];
}
exports.getRarePokemon = getRarePokemon;

var getNotCommonPokemon = function(n)
{
    return not_common[n];
}
exports.getNotCommonPokemon = getNotCommonPokemon;

var getCommonPokemon = function(n)
{
    return common[n];
}
exports.getCommonPokemon = getCommonPokemon;

var getVeryCommonPokemon = function(n)
{
    return very_common[n];
}
exports.getVeryCommonPokemon = getVeryCommonPokemon;

var ultimate = [
    {  nom: '', id: '1', png: '../images/pokemons/png/06_ultimate/1.png', gif: 'pokemons/gif/1.gif' },
    {  nom: '', id: '2', png: '../images/pokemons/png/06_ultimate/2.png', gif: '' }
];
exports.ultimate = ultimate;

var very_rare = [
    {  nom: '', id: '1', png: '../images/pokemons/png/05_very_rare/1.png', gif: '' },
    {  nom: '', id: '2', png: '../images/pokemons/png/05_very_rare/2.png', gif: '' },
    {  nom: '', id: '3', png: '../images/pokemons/png/05_very_rare/3.png', gif: '' },
    {  nom: '', id: '4', png: '../images/pokemons/png/05_very_rare/4.png', gif: '' },
    {  nom: '', id: '5', png: '../images/pokemons/png/05_very_rare/5.png', gif: '' },
    {  nom: '', id: '6', png: '../images/pokemons/png/05_very_rare/6.png', gif: '' },
    {  nom: '', id: '7', png: '../images/pokemons/png/05_very_rare/7.png', gif: '' },
    {  nom: '', id: '8', png: '../images/pokemons/png/05_very_rare/8.png', gif: '' },
    {  nom: '', id: '9', png: '../images/pokemons/png/05_very_rare/9.png', gif: '' },
    {  nom: '', id: '10', png: '../images/pokemons/png/05_very_rare/10.png', gif: '' },
    {  nom: '', id: '11', png: '../images/pokemons/png/05_very_rare/11.png', gif: '' },
    {  nom: '', id: '12', png: '../images/pokemons/png/05_very_rare/12.png', gif: '' },
    {  nom: '', id: '13', png: '../images/pokemons/png/05_very_rare/13.png', gif: '' },
    {  nom: '', id: '14', png: '../images/pokemons/png/05_very_rare/14.png', gif: '' },
    {  nom: '', id: '15', png: '../images/pokemons/png/05_very_rare/15.png', gif: '' },
    {  nom: '', id: '16', png: '../images/pokemons/png/05_very_rare/16.png', gif: '' },
    {  nom: '', id: '17', png: '../images/pokemons/png/05_very_rare/17.png', gif: '' },
    {  nom: '', id: '18', png: '../images/pokemons/png/05_very_rare/18.png', gif: '' },
    {  nom: '', id: '19', png: '../images/pokemons/png/05_very_rare/19.png', gif: '' },
    {  nom: '', id: '20', png: '../images/pokemons/png/05_very_rare/20.png', gif: '' },
    {  nom: '', id: '21', png: '../images/pokemons/png/05_very_rare/21.png', gif: '' },
    {  nom: '', id: '22', png: '../images/pokemons/png/05_very_rare/22.png', gif: '' },
    {  nom: '', id: '23', png: '../images/pokemons/png/05_very_rare/23.png', gif: '' },
    {  nom: '', id: '24', png: '../images/pokemons/png/05_very_rare/24.png', gif: '' },
    {  nom: '', id: '25', png: '../images/pokemons/png/05_very_rare/25.png', gif: '' },
    {  nom: '', id: '26', png: '../images/pokemons/png/05_very_rare/26.png', gif: '' }
];
exports.very_rare = very_rare;

var rare = [
    {  nom: '', id: '1', png: '../images/pokemons/png/04_rare/1.png', gif: '' },
    {  nom: '', id: '2', png: '../images/pokemons/png/04_rare/2.png', gif: '' },
    {  nom: '', id: '3', png: '../images/pokemons/png/04_rare/3.png', gif: '' },
    {  nom: '', id: '4', png: '../images/pokemons/png/04_rare/4.png', gif: '' },
    {  nom: '', id: '5', png: '../images/pokemons/png/04_rare/5.png', gif: '' },
    {  nom: '', id: '6', png: '../images/pokemons/png/04_rare/6.png', gif: '' },
    {  nom: '', id: '7', png: '../images/pokemons/png/04_rare/7.png', gif: '' },
    {  nom: '', id: '8', png: '../images/pokemons/png/04_rare/8.png', gif: '' },
    {  nom: '', id: '9', png: '../images/pokemons/png/04_rare/9.png', gif: '' },
    {  nom: '', id: '10', png: '../images/pokemons/png/04_rare/10.png', gif: '' },
    {  nom: '', id: '11', png: '../images/pokemons/png/04_rare/11.png', gif: '' },
    {  nom: '', id: '12', png: '../images/pokemons/png/04_rare/12.png', gif: '' },
    {  nom: '', id: '13', png: '../images/pokemons/png/04_rare/13.png', gif: '' },
    {  nom: '', id: '14', png: '../images/pokemons/png/04_rare/14.png', gif: '' },
    {  nom: '', id: '15', png: '../images/pokemons/png/04_rare/15.png', gif: '' },
    {  nom: '', id: '16', png: '../images/pokemons/png/04_rare/16.png', gif: '' },
    {  nom: '', id: '17', png: '../images/pokemons/png/04_rare/17.png', gif: '' },
    {  nom: '', id: '18', png: '../images/pokemons/png/04_rare/18.png', gif: '' },
    {  nom: '', id: '19', png: '../images/pokemons/png/04_rare/19.png', gif: '' },
    {  nom: '', id: '20', png: '../images/pokemons/png/04_rare/20.png', gif: '' },
    {  nom: '', id: '21', png: '../images/pokemons/png/04_rare/21.png', gif: '' },
    {  nom: '', id: '22', png: '../images/pokemons/png/04_rare/22.png', gif: '' },
    {  nom: '', id: '23', png: '../images/pokemons/png/04_rare/23.png', gif: '' },
    {  nom: '', id: '24', png: '../images/pokemons/png/04_rare/24.png', gif: '' },
    {  nom: '', id: '25', png: '../images/pokemons/png/04_rare/25.png', gif: '' },
    {  nom: '', id: '26', png: '../images/pokemons/png/04_rare/26.png', gif: '' },
    {  nom: '', id: '27', png: '../images/pokemons/png/04_rare/27.png', gif: '' },
    {  nom: '', id: '28', png: '../images/pokemons/png/04_rare/28.png', gif: '' },
    {  nom: '', id: '29', png: '../images/pokemons/png/04_rare/29.png', gif: '' },
    {  nom: '', id: '30', png: '../images/pokemons/png/04_rare/30.png', gif: '' },
    {  nom: '', id: '31', png: '../images/pokemons/png/04_rare/31.png', gif: '' },
    {  nom: '', id: '32', png: '../images/pokemons/png/04_rare/32.png', gif: '' }
];
exports.rare = rare;

var not_common = [
    {  nom: '', id: '1', png: '../images/pokemons/png/03_not_common/1.png', gif: '' },
    {  nom: '', id: '2', png: '../images/pokemons/png/03_not_common/2.png', gif: '' },
    {  nom: '', id: '3', png: '../images/pokemons/png/03_not_common/3.png', gif: '' },
    {  nom: '', id: '4', png: '../images/pokemons/png/03_not_common/4.png', gif: '' },
    {  nom: '', id: '5', png: '../images/pokemons/png/03_not_common/5.png', gif: '' },
    {  nom: '', id: '6', png: '../images/pokemons/png/03_not_common/6.png', gif: '' },
    {  nom: '', id: '7', png: '../images/pokemons/png/03_not_common/7.png', gif: '' },
    {  nom: '', id: '8', png: '../images/pokemons/png/03_not_common/8.png', gif: '' },
    {  nom: '', id: '9', png: '../images/pokemons/png/03_not_common/9.png', gif: '' },
    {  nom: '', id: '10', png: '../images/pokemons/png/03_not_common/10.png', gif: '' },
    {  nom: '', id: '11', png: '../images/pokemons/png/03_not_common/11.png', gif: '' },
    {  nom: '', id: '12', png: '../images/pokemons/png/03_not_common/12.png', gif: '' },
    {  nom: '', id: '13', png: '../images/pokemons/png/03_not_common/13.png', gif: '' },
    {  nom: '', id: '14', png: '../images/pokemons/png/03_not_common/14.png', gif: '' },
    {  nom: '', id: '15', png: '../images/pokemons/png/03_not_common/15.png', gif: '' },
    {  nom: '', id: '16', png: '../images/pokemons/png/03_not_common/16.png', gif: '' },
    {  nom: '', id: '17', png: '../images/pokemons/png/03_not_common/17.png', gif: '' },
    {  nom: '', id: '18', png: '../images/pokemons/png/03_not_common/18.png', gif: '' },
    {  nom: '', id: '19', png: '../images/pokemons/png/03_not_common/19.png', gif: '' },
    {  nom: '', id: '20', png: '../images/pokemons/png/03_not_common/20.png', gif: '' },
    {  nom: '', id: '21', png: '../images/pokemons/png/03_not_common/21.png', gif: '' },
    {  nom: '', id: '22', png: '../images/pokemons/png/03_not_common/22.png', gif: '' },
    {  nom: '', id: '23', png: '../images/pokemons/png/03_not_common/23.png', gif: '' },
    {  nom: '', id: '24', png: '../images/pokemons/png/03_not_common/24.png', gif: '' },
    {  nom: '', id: '25', png: '../images/pokemons/png/03_not_common/25.png', gif: '' },
    {  nom: '', id: '26', png: '../images/pokemons/png/03_not_common/26.png', gif: '' },
    {  nom: '', id: '27', png: '../images/pokemons/png/03_not_common/27.png', gif: '' },
    {  nom: '', id: '28', png: '../images/pokemons/png/03_not_common/28.png', gif: '' },
    {  nom: '', id: '29', png: '../images/pokemons/png/03_not_common/29.png', gif: '' },
    {  nom: '', id: '30', png: '../images/pokemons/png/03_not_common/30.png', gif: '' }
];
exports.not_common = not_common;

var common = [
    {  nom: '', id: '1', png: '../images/pokemons/png/02_common/1.png', gif: '' },
    {  nom: '', id: '2', png: '../images/pokemons/png/02_common/2.png', gif: '' },
    {  nom: '', id: '3', png: '../images/pokemons/png/02_common/3.png', gif: '' },
    {  nom: '', id: '4', png: '../images/pokemons/png/02_common/4.png', gif: '' },
    {  nom: '', id: '5', png: '../images/pokemons/png/02_common/5.png', gif: '' },
    {  nom: '', id: '6', png: '../images/pokemons/png/02_common/6.png', gif: '' },
    {  nom: '', id: '7', png: '../images/pokemons/png/02_common/7.png', gif: '' },
    {  nom: '', id: '8', png: '../images/pokemons/png/02_common/8.png', gif: '' },
    {  nom: '', id: '9', png: '../images/pokemons/png/02_common/9.png', gif: '' },
    {  nom: '', id: '10', png: '../images/pokemons/png/02_common/10.png', gif: '' },
    {  nom: '', id: '11', png: '../images/pokemons/png/02_common/11.png', gif: '' },
    {  nom: '', id: '12', png: '../images/pokemons/png/02_common/12.png', gif: '' },
    {  nom: '', id: '13', png: '../images/pokemons/png/02_common/13.png', gif: '' },
    {  nom: '', id: '14', png: '../images/pokemons/png/02_common/14.png', gif: '' },
    {  nom: '', id: '15', png: '../images/pokemons/png/02_common/15.png', gif: '' },
    {  nom: '', id: '16', png: '../images/pokemons/png/02_common/16.png', gif: '' },
    {  nom: '', id: '17', png: '../images/pokemons/png/02_common/17.png', gif: '' },
    {  nom: '', id: '18', png: '../images/pokemons/png/02_common/18.png', gif: '' },
    {  nom: '', id: '19', png: '../images/pokemons/png/02_common/19.png', gif: '' },
    {  nom: '', id: '20', png: '../images/pokemons/png/02_common/20.png', gif: '' },
    {  nom: '', id: '21', png: '../images/pokemons/png/02_common/21.png', gif: '' },
    {  nom: '', id: '22', png: '../images/pokemons/png/02_common/22.png', gif: '' },
    {  nom: '', id: '23', png: '../images/pokemons/png/02_common/23.png', gif: '' },
    {  nom: '', id: '24', png: '../images/pokemons/png/02_common/24.png', gif: '' },
    {  nom: '', id: '25', png: '../images/pokemons/png/02_common/25.png', gif: '' },
    {  nom: '', id: '26', png: '../images/pokemons/png/02_common/26.png', gif: '' },
    {  nom: '', id: '27', png: '../images/pokemons/png/02_common/27.png', gif: '' },
    {  nom: '', id: '28', png: '../images/pokemons/png/02_common/28.png', gif: '' },
    {  nom: '', id: '29', png: '../images/pokemons/png/02_common/29.png', gif: '' },
    {  nom: '', id: '30', png: '../images/pokemons/png/02_common/30.png', gif: '' }
];
exports.common = common;

var very_common = [
    {  nom: '', id: '1', png: '../images/pokemons/png/01_very_common/1.png', gif: '' },
    {  nom: '', id: '2', png: '../images/pokemons/png/01_very_common/2.png', gif: '' },
    {  nom: '', id: '3', png: '../images/pokemons/png/01_very_common/3.png', gif: '' },
    {  nom: '', id: '4', png: '../images/pokemons/png/01_very_common/4.png', gif: '' },
    {  nom: '', id: '5', png: '../images/pokemons/png/01_very_common/5.png', gif: '' },
    {  nom: '', id: '6', png: '../images/pokemons/png/01_very_common/6.png', gif: '' },
    {  nom: '', id: '7', png: '../images/pokemons/png/01_very_common/7.png', gif: '' },
    {  nom: '', id: '8', png: '../images/pokemons/png/01_very_common/8.png', gif: '' },
    {  nom: '', id: '9', png: '../images/pokemons/png/01_very_common/9.png', gif: '' },
    {  nom: '', id: '10', png: '../images/pokemons/png/01_very_common/10.png', gif: '' },
    {  nom: '', id: '11', png: '../images/pokemons/png/01_very_common/11.png', gif: '' },
    {  nom: '', id: '12', png: '../images/pokemons/png/01_very_common/12.png', gif: '' },
    {  nom: '', id: '13', png: '../images/pokemons/png/01_very_common/13.png', gif: '' },
    {  nom: '', id: '14', png: '../images/pokemons/png/01_very_common/14.png', gif: '' },
    {  nom: '', id: '15', png: '../images/pokemons/png/01_very_common/15.png', gif: '' },
    {  nom: '', id: '16', png: '../images/pokemons/png/01_very_common/16.png', gif: '' },
    {  nom: '', id: '17', png: '../images/pokemons/png/01_very_common/17.png', gif: '' },
    {  nom: '', id: '18', png: '../images/pokemons/png/01_very_common/18.png', gif: '' },
    {  nom: '', id: '19', png: '../images/pokemons/png/01_very_common/19.png', gif: '' },
    {  nom: '', id: '20', png: '../images/pokemons/png/01_very_common/20.png', gif: '' },
    {  nom: '', id: '21', png: '../images/pokemons/png/01_very_common/21.png', gif: '' },
    {  nom: '', id: '22', png: '../images/pokemons/png/01_very_common/22.png', gif: '' },
    {  nom: '', id: '23', png: '../images/pokemons/png/01_very_common/23.png', gif: '' },
    {  nom: '', id: '24', png: '../images/pokemons/png/01_very_common/24.png', gif: '' },
    {  nom: '', id: '25', png: '../images/pokemons/png/01_very_common/25.png', gif: '' },
    {  nom: '', id: '26', png: '../images/pokemons/png/01_very_common/26.png', gif: '' },
    {  nom: '', id: '27', png: '../images/pokemons/png/01_very_common/27.png', gif: '' },
    {  nom: '', id: '28', png: '../images/pokemons/png/01_very_common/28.png', gif: '' },
    {  nom: '', id: '29', png: '../images/pokemons/png/01_very_common/29.png', gif: '' },
    {  nom: '', id: '30', png: '../images/pokemons/png/01_very_common/30.png', gif: '' },
    {  nom: '', id: '31', png: '../images/pokemons/png/01_very_common/31.png', gif: '' }
];
exports.very_common = very_common;
